tests/bor/bor_api_test.go

```diff
@@ -1,168 +1,213 @@
+
+				// write the state-sync receipts to database => receipts[i][1:] => receipts[i][1]
+				// State sync logs don't have tx index, tx hash and other necessary fields, DeriveFieldsForBorLogs will fill those fields for websocket subscriptions
+				// DeriveFieldsForBorLogs argurments:
+				// 1. State-sync logs
+				// 2. Block Hash
+				// 3. Block Number
+				// 4. Transactions in the block(except state-sync) i.e. 1 in our case
+				// 5. AllLogs -(minus) StateSyncLogs ; since we only have state-sync tx, it will be 1
+				types.DeriveFieldsForBorLogs(receipts[i][1].Logs, block.Hash(), block.NumberU64(), uint(1), uint(1))
+
+				rawdb.WriteBorReceipt(blockBatch, block.Hash(), block.NumberU64(), &types.ReceiptForStorage{
+					Status: types.ReceiptStatusSuccessful, // make receipt status successful
+					Logs:   receipts[i][1].Logs,
+				})
+
+				rawdb.WriteBorTxLookupEntry(blockBatch, block.Hash(), block.NumberU64())
 
-	for n := 0; n < int(spanSize)+1; n++ {
-		rpcNumber := rpc.BlockNumberOrHashWithNumber(rpc.BlockNumber(n))
+			}
 
-		txs, err := ethAPI.GetTransactionReceiptsByBlock(context.Background(), rpcNumber)
-		require.Nil(t, err)
+		}
 
-		tx := txPoolAPI.GetTransactionByBlockNumberAndIndex(context.Background(), rpc.BlockNumber(n), 0)
+		if err := blockBatch.Write(); err != nil {
+			fmt.Println("Failed to write block into disk", "err", err)
+		}
+	}
 
-		blockMap, err := ethAPI.GetBlockByNumber(context.Background(), rpc.BlockNumber(n), true)
-		require.Nil(t, err)
+	publicBlockchainAPI := backend.PublicBlockChainAPI()
 
-		expectedTxHash, ok := txHashes[n]
-		// FIXME: add `IsSprintStart(uint64(n)) || IsSpanStart(uint64(n))` after adding a full state receiver contract
-		if ok {
-			require.Len(t, txs, 1)
+	// check 1 : zero transactions
+	receiptsOut, err := publicBlockchainAPI.GetTransactionReceiptsByBlock(context.Background(), rpc.BlockNumberOrHashWithNumber(1))
+	if err != nil {
+		fmt.Println(err)
+	}
 
-			require.NotNil(t, tx, "not nil receipt expected")
+	assert.Equal(t, 0, len(receiptsOut))
 
-			require.Equal(t, expectedTxHash, tx.Hash, "got different from expected receipt")
+	// check 2 : one transactions ( normal )
+	receiptsOut, err = publicBlockchainAPI.GetTransactionReceiptsByBlock(context.Background(), rpc.BlockNumberOrHashWithNumber(2))
+	if err != nil {
+		fmt.Println(err)
+	}
 
-			blockTxs, ok := blockMap["transactions"].([]interface{})
-			require.Len(t, blockTxs, 1)
+	assert.Equal(t, 1, len(receiptsOut))
+	assert.True(t, areDifferentHashes(receiptsOut))
 
-			blockTx, ok := blockTxs[0].(*ethapi.RPCTransaction)
-			require.True(t, ok)
-			require.Equal(t, expectedTxHash, blockTx.Hash)
-		} else {
-			require.Len(t, txs, 0)
+	// check 3 : two transactions ( both normal )
+	receiptsOut, err = publicBlockchainAPI.GetTransactionReceiptsByBlock(context.Background(), rpc.BlockNumberOrHashWithNumber(3))
+	if err != nil {
+		fmt.Println(err)
+	}
```

Comment:
Debug?

Reply:
Yes, my bad. Thank you for pointing out 💜